File: VendorCodeArchive/TestFiles/SemgrepTestFile.cs
using System;
using System.Data.SqlClient;
using System.Diagnostics;
using System.Security.Cryptography;

namespace VendorCodeArchive.TestFiles
{
    /// <summary>
    /// Test file for Semgrep plugin validation
    /// Contains intentional security violations to test semgrep-validator plugin
    /// </summary>
    public class SemgrepTestFile
    {
        // VIOLATION: Hardcoded API key (SEMGREP-SEC-HARDCODE)
        private const string ApiKey = "FAKE_API_KEY_1234567890_DO_NOT_USE";

        // VIOLATION: Hardcoded database password (SEMGREP-SEC-HARDCODE)
        private const string DbPassword = "HARDCODED_PASSWORD_FOR_TESTING";

        /// <summary>
        /// VIOLATION: SQL Injection vulnerability (SEMGREP-SEC-SQLINJEC)
        /// </summary>
        public void UnsafeDatabaseQuery(string userId)
        {
            var connectionString = $"Server=localhost;Database=TestDB;User Id=admin;Password={DbPassword};";
            using (var connection = new SqlConnection(connectionString))
            {
                connection.Open();

                // SQL Injection: user input directly concatenated into SQL query
                var query = "SELECT * FROM Users WHERE UserId = '" + userId + "'";
                using (var command = new SqlCommand(query, connection))
                {
                    var result = command.ExecuteScalar();
                }
            }
        }

        /// <summary>
        /// VIOLATION: Command Injection vulnerability (SEMGREP-SEC-COMMANDI)
        /// </summary>
        public void UnsafeCommandExecution(string filename)
        {
            // Command injection: user input passed directly to process
            var process = new Process
            {
                StartInfo = new ProcessStartInfo
                {
                    FileName = "cmd.exe",
                    Arguments = "/c type " + filename,  // Unsafe concatenation
                    UseShellExecute = false
                }
            };
            process.Start();
        }

        /// <summary>
        /// VIOLATION: Weak cryptography (SEMGREP-SEC-WEAKCRYP)
        /// </summary>
        public string WeakHash(string input)
        {
            // MD5 is cryptographically broken
            using (var md5 = MD5.Create())
            {
                var hash = md5.ComputeHash(System.Text.Encoding.UTF8.GetBytes(input));
                return BitConverter.ToString(hash).Replace("-", "").ToLowerInvariant();
            }
        }

        /// <summary>
        /// VIOLATION: XSS vulnerability in web context (SEMGREP-SEC-XSS)
        /// </summary>
        public string UnsafeHtmlGeneration(string userInput)
        {
            // User input not sanitized before rendering in HTML
            return $"<div>Welcome, {userInput}!</div>";
        }

        /// <summary>
        /// VIOLATION: Hardcoded encryption key (SEMGREP-SEC-HARDCODE)
        /// </summary>
        public byte[] GetEncryptionKey()
        {
            // Hardcoded encryption key - should be from secure key store
            return new byte[] { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08 };
        }
    }
}

File: VendorCodeArchive/TestFiles/SemgrepTestFile.py
"""
Test file for Semgrep plugin validation
Contains intentional security violations to test semgrep-validator plugin
"""

import hashlib
import sqlite3
from flask import request

# VIOLATION: SQL Injection (SEMGREP-SEC-SQLINJEC)
def unsafe_database_query():
    """SQL injection vulnerability - string concatenation"""
    user_id = request.args.get('user_id')
    conn = sqlite3.connect('database.db')

    # SQL Injection: user input directly concatenated into SQL query
    query = "SELECT * FROM users WHERE user_id = '" + user_id + "'"
    conn.execute(query)

    # Another SQL injection variant
    query2 = "SELECT * FROM orders WHERE id = %s" % (user_id,)
    conn.execute(query2)

    # SQL injection with format
    query3 = "DELETE FROM sessions WHERE token = {}".format(user_id)
    conn.execute(query3)

    # SQL injection with f-string
    query4 = f"UPDATE users SET active = 1 WHERE id = {user_id}"
    conn.execute(query4)


# VIOLATION: Weak Cryptography (SEMGREP-SEC-WEAKCRYP)
def weak_hash_function(password):
    """Using MD5 for password hashing - cryptographically broken"""
    # MD5 is not secure for password hashing
    hashed = hashlib.md5(password.encode()).hexdigest()
    return hashed


# VIOLATION: Hardcoded secrets (would match regex rules if any)
API_KEY = "FAKE_API_KEY_FOR_TESTING_DO_NOT_USE"
DATABASE_PASSWORD = "hardcoded_password_123"


# VIOLATION: Another SQL injection variant
def another_sql_injection():
    """SQL injection with variable concatenation"""
    user_input = request.form.get('search')
    conn = sqlite3.connect('database.db')

    # Build query with concatenation
    sql = "SELECT * FROM products WHERE name = '" + user_input + "'"
    conn.execute(sql)


# VIOLATION: Command injection possibility
def unsafe_command_execution():
    """Potential command injection if user input reaches os.system"""
    import os
    filename = request.args.get('file')
    # This would be command injection if executed
    os.system("cat " + filename)
